src/api/useFetch.js

```diff
@@ -19,6 +19,65 @@ import { FeatureRegistrationContext } from '../App'
 import { registerCallbackHandler } from '../feature'
 
 
+const useFetch = (
+  route,
+  params,
+  errorMsg,
+) => {
+  const [res, setRes] = React.useState()
+  const [isLoading, setIsLoading] = React.useState(true)
+  const [isError, setIsError] = React.useState()
+  const [error, setError] = React.useState()
+  const [locked, setLocked] = React.useState(false)
```

Comment:
should we use `useEffect` rather than locking ourselves?

Reply:
we use this locking mechanism to prevent multiple, redundant requests
they would occur if the component using our useFetch hook is re-rendered

how could we achieve the same semantics with useEffect?

Reply:
hmmm, I see. at the very least, this might be worth a comment / docstr, then